src/linear_algebra/bilinear_form.lean

```diff
@@ -470,9 +477,9 @@ def is_pair_self_adjoint (f : module.End R M) := is_adjoint_pair B B' f f
 /-- The set of pair-self-adjoint endomorphisms are a submodule of the type of all endomorphisms. -/
 def is_pair_self_adjoint_submodule : submodule R (module.End R M) :=
 { carrier := { f | is_pair_self_adjoint B B' f },
-  zero    := is_adjoint_pair_zero B B',
-  add     := λ f g hf hg, is_adjoint_pair_add B B' _ _ _ _ hf hg,
-  smul    := λ c f h, is_adjoint_pair_smul B B' _ _ c h, }
+  zero    := is_adjoint_pair_zero,
+  add     := λ f g hf hg, is_adjoint_pair.add hf hg,
```

Comment:
You should use your new powers :tada: :
```suggestion
  add     := λ f g hf hg, hf.add hg,
```
similar elsewhere

Reply:
Or is this not working, because `hf` is not exactly an `is_adjoint_pair`? What if you make all the "special case" defs into `abbreviation`s?

Reply:
I'll take a pass over it again to see where else this pattern might help.

Reply:
Now done. I have avoided shortening expressions of the form `bilin_form.is_adjoint_pair B B'` to `B.is_adjoint_pair B'` since I feel this breaks symmetry a bit too much but it's not a  strong feeling.